src/Microsoft.DotNet.Interactive.Http.Parsing/Parsing/HttpRequestParser.cs

```diff
@@ -541,6 +551,13 @@ private bool IsAtStartOfEmbeddedExpression() =>
             CurrentToken is { Text: "{" } &&
             CurrentTokenPlus(1) is { Text: "{" };
 
+        private bool IsStartOfEscapedCharacterSequence() =>
+            CurrentToken is { Kind: TokenKind.Punctuation } and { Text: "\\" } &&
+            (CurrentTokenPlus(1) is { Kind: TokenKind.Punctuation } and { Text: "{" } &&
+            CurrentTokenPlus(2) is { Kind: TokenKind.Punctuation } and { Text: "{" }) ||
+            (CurrentTokenPlus(1) is { Kind: TokenKind.Punctuation } and { Text: "}" } &&
```

Comment:
This looks unintuitive. This is really the end of an escape sequence but only if the preceding opening `{{` was preceded with "\".